### PR TITLE
First version

### PR DESCRIPTION
**Pull request by jhford**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,73 @@
...
+
+* Automatically generate source maps with correct file references
+* Cleans output directory
+* Allows us to load configuration from an NPM module instead of copying around a .babelrc
+
+## Getting started
+First, you're going to want to install this package and the taskcluster global configs
+```
+npm install taskcluster-babel --save-dev
+npm install taskcluster-configs --save-dev
```
Comment (jonasfj) [inline README.md:19]: Can we call this for `taskcluster-lib-rules` as it is a library of rules... roles for eslint, rules for babel, I'm not sure we'll have more... but that would help...

Calling it configs could be slightly confusing...

**Comment by jonasfj** [comment]
hmm... How about we call this something completely unrelated to taskcluster... It doesn't really have anything to do with our platform other than we use it.

How about a name like `simple-babel-cli`, `sane-babel-cli` or `johns-babel-cli` :)
IMO we should try to make generic modules whenever possible, calling things `taskcluster-` is not always ideal. It's good for tc specific things.

File: lib/tc-babel.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        } else {
+          fs.unlinkSync(fullPath);
+        }
+      });
+      fs.rmdirSync(dir);
+    }
+  } else {
+    fs.unlinkSync(dir);
+  }
+}
```
Comment (jonasfj) [inline lib/tc-babel.js:37]: nit, but there is also `rmrf` on npm :)

(probably I should find a list folder function on npm)

File: package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "taskcluster-babel",
+  "version": "0.0.1",
+  "description": "babel executable for taskcluster projects",
+  "main": "lib/tc-babel.js",
+  "scripts": {
+    "test": "mocha test/*_test.js"
+  },
+  "bin": {
+    "tc-babel": "./bin/cli.js"
```
Comment (jonasfj) [inline package.json:10]: so, calling the package `sane-babel-cli` and this `sane-babel` would be fine :)

Also you forgot to `git add bin/cli.js`

File: package.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/taskcluster/taskcluster-babel.git"
+  },
+  "keywords": [
+    "taskcluster",
+    "babel"
+  ],
+  "author": "John Ford <jhford.mozilla@com>",
```
Comment (jonasfj) [inline package.json:20]: I have to ask what is up with your email?

File: lib/tc-babel.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    assert(typeof src === 'string', 'source directory must be string');
+    assert(typeof dst === 'string', 'destination directory must be string');
+    assert(fs.existsSync(src), 'source directory must exist');
+    if (fs.existsSync(dst)) {
+      console.log('Deleting existing destintation: ' + dst);
+      deleteSync(dst);
+      fs.mkdirSync(dst);
+    } else {
+      fs.mkdirSync(dst);
+    }
```
Comment (jonasfj) [inline lib/tc-babel.js:59]: Nit but drop the `else` and move fs.mkdirSync a level up...

**Comment by jonasfj** [comment]
Generally the code looks nice... missing `bin/cli.js`, and would really love a rename.

File: lib/tc-babel.js (hunk 2 above)
Comment (gregarndt) [inline lib/tc-babel.js:37]: I use https://www.npmjs.com/package/rimraf with good luck.

File: package.json (hunk 4 above)
Comment (jhford) [inline package.json:20]: spambots can suck it!

File: lib/tc-babel.js (hunk 2 above)
Comment (jonasfj) [inline lib/tc-babel.js:37]: oh, yeah that's the good package :)

File: package.json (hunk 4 above)
Comment (jonasfj) [inline package.json:20]: lol figured it might be intentional...

**Comment by jonasfj** [comment]
@jhford, I propose a rename to `jhfords-babel-cli` or `simple-babel-cli`, but feel free to merge this is see no issues with it.

I just think we should try to call it something that doesn't discourage others from using the cli too... The concept is solid, and we can't be the only ones interested in this.

**Pull request closed without merging**